### ISSUE TITLE
Side-data loading scripts needlessly require setting the working directory

### ISSUE THREAD

**Issue by fedde-s** [cBioPortal/cbioportal#6123]
The .pl (Perl) entrypoints that form the documented ways to load data outside of studies, such as gene symbols and aliases and gene sets, only work if you change the working directory to `cbioportal_source/core/src/main/scripts`, the folder containing those scripts.

Removing this requirement, either by changing the way they find `env.pl` or replacing these one-line wrappers altogether, would improve usability. It would for instance allow installing symlinks to `$PATH`, or running the scripts by full path while the working directory is the folder containing the data files.

**Comment by jjgao** [comment]
I don't really understand why we still need the Perl code. It is possible to let the Python code to handle everything?

**Comment by fedde-s** [comment]
It would be, if we add a few lines of Python code and change the documentation that refers to the Perl scripts. The Perl scripts do very little except check some environment variables and start the JVM with the right arguments.

**Comment by jjgao** [comment]
@fedde-s I think we should do that and get rid of the Perl package. Would you or someone at Hyve be able to take of that? @Sjoerd-van-Hagen

**Comment by Sjoerd-van-Hagen** [comment]
@jjgao we are swamped. Who included this Perl package?

**Comment by jjgao** [comment]
The Perl package is old than me... I don't think it's urgent to get rid of it. Maybe put it on your back burner?

**Comment by fedde-s** [comment]
Given who's currently working on what in our team, I think @SRodenburg or @oplantalech would be the most likely to have the opportunity to resolve this. Maybe @pvannierop.

**Comment by Sjoerd-van-Hagen** [comment]
@jjgao why do you assign this to us?

**Comment by jjgao** [comment]
@Sjoerd-van-Hagen I think you are the maintainer of the python code and we should replace the perl code with python code...

**Comment by Sjoerd-van-Hagen** [comment]
@jjgao I had a quick look at the .pl scripts in that dir. It seems most of them or so old that I highly doubt if they still work as they should. I also highly doubt if anyone outside of MSK is using these as they are not documented I think. The only scripts that is used often would be the migrate database one.

Can you check which scripts can simply be thrown away?

**Comment by jjgao** [comment]
Strange, I couldn't find any code/documentation that is using env.pl that @fedde-s reported above. I found a couple of other places using the perl code.

importGenePanel.pl in https://docs.cbioportal.org/2.1.2-deploy-without-docker/load-sample-cancer-study

importGenesetData.pl https://docs.cbioportal.org/5.1-data-loading/data-loading/import-gene-sets

I think we should get rid of all the perl code.

@sheridancbio do you know if we are using any of the perl code at MSK?

**Comment by Sjoerd-van-Hagen** [comment]
@jjgao the env.pl is imported into many of the .pl scripts. 

importGenesetHierarchy.pl is also mentioned in  https://docs.cbioportal.org/5.1-data-loading/data-loading/import-gene-sets

Seems like we were the only ones adding the scripts we wrote to the documentation. We copied your approach when creating them. All the scripts that are of this format can be translated into Bash.

There are also some .pl scripts that are more involved. I think a first test would be to run the scripts and see if they still work, if not they can be deleted immediately. We can then decide how we deal with the rest.
